src/Data/EuclideanRing.purs

```diff
@@ -69,3 +73,17 @@ foreign import intDiv :: Int -> Int -> Int
 foreign import intMod :: Int -> Int -> Int
 
 foreign import numDiv :: Number -> Number -> Number
+
+-- | The *greatest common divisor* of two values.
+gcd :: forall a. (Eq a, EuclideanRing a) => a -> a -> a
+gcd a b =
+  if b == zero
+    then a
+    else gcd b (a `mod` b)
+
+-- | The *least common multiple* of two values.
+lcm :: forall a. (Eq a, EuclideanRing a) => a -> a -> a
+lcm a b =
+  if a == zero && b == zero
```

Comment:
I think it can, and actually it sounds like that would be better, according to the first paragraph of https://en.wikipedia.org/wiki/Least_common_multiple:

> In arithmetic and number theory, the least common multiple, lowest common multiple, or smallest common multiple of two integers a and b, usually denoted by LCM(a, b), is the smallest positive integer that is divisible by both a and b.[1] Since division of integers by zero is undefined, this definition has meaning only if a and b are both different from zero.[2] However, some authors define lcm(a,0) as 0 for all a, which is the result of taking the lcm to be the least upper bound in the lattice of divisibility.